site/content/integrate/apps/interactivity/dynamic-form-example/_index.md

```diff
@@ -0,0 +1,544 @@
+
+Note that the `expand` object with `post: 'summary'` means that we will receive the relevant `Post` at `call.context.post` when we receive the request. This makes it so we don't have to fetch the `Post` manually when handling the request. Learn more about Expand [here]({{< ref "call" >}}).
+
+By defining the post menu item's `Call` with the path `/forms/create-ticket-post-menu`, we've implicitly registered an endpoint to process a request when this item is clicked:
+- `/forms/create-ticket-post-menu/submit`
+
+### Opening a modal
+
+Now let's process the post menu item click, and return an `AppForm` to show as a modal:
+
+```ts
+app.post('/forms/create-ticket-post-menu/submit', (req, res) => {
+    const call: AppCallRequest = req.body;
+
+    // Extract the Post from the expanded context
+    const post: Post = call.context.post;
+
+    const call: AppCall = {
+        path: '/forms/create-ticket',
+        expand: {
+            acting_user: 'summary', // Expand the acting user so we can get the user's username for our post
+            team: 'summary', // Expand the current team so we can construct a permalink for the selected post using the team's name
+        },
+    };
+
+    const form: AppForm = {
+        title: 'Create a Ticket',
+        fields: [
+            {
+                name: 'summary',
+                value: '',
+                modal_label: 'Summary',
+                type: 'text',
+                is_required: true, // This makes it so the user must provide this value before submitting the form
+                min_length: 10, // We set a minimum length of 10 characters for this field
+            },
+            {
+                name: 'description',
+                value: post.message, // Use the Post's message as the default value for the description field
+                modal_label: 'Description',
+                type: 'text',
+                subtype: 'textarea',
+            },
+        ],
+    };
+
+    const callResponse: AppCallResponse = {
+        type: 'form',
+        call,
+        form,
+    }
+
+    res.json(callResponse);
+});
+```
+
+The user is shown a form with text fields labeled "Summary" and "Description". The Description field is prepopulated with the `Post`'s message.
+
+By defining the form's `Call` with the path `/forms/create-ticket`, we've implicitly registered three endpoints for this form to use:
+- `/forms/create-ticket/lookup` is used to fetch autocomplete values
+- `/forms/create-ticket/form` is used to dynamically redefine the modal when a particular field's value is chosen
+- `/forms/create-ticket/submit` is used to process the form's submission
+
+### Processing the form submission
+
+Now let's process the form submitted at the path `/forms/create-ticket/submit`
+
+We've specified in the `expand` clause to include the acting `User` that is submitting the form, as well as the `Team` so we can construct a post's permalink using the team's name.
```

Comment:
" so we can construct a post's permalink using the team's name." - Might make sense to specify/state this goal at the beginning (as a sub-goal of the ticket creation main goal)?

Reply:
I would even rephrase this to change the order. Something like:

In the ticket we are going to create, we want to add the permalink to this post. A permalink has the shape `baseURL/teamname/post/postID` (or something like that). To get the team name, we have added the team as a call expansion.